core/templates/dev/head/pages/state_editor/StateInteractionEditorDirective.js

```diff
@@ -213,8 +215,21 @@ oppia.directive('stateInteractionEditor', [
                       UrlInterpolationService.getInteractionThumbnailImageUrl);
 
                     $scope.INTERACTION_SPECS = INTERACTION_SPECS;
-                    $scope.ALLOWED_INTERACTION_CATEGORIES = (
-                      GLOBALS.ALLOWED_INTERACTION_CATEGORIES);
+
+                    // Checks that the modal is invoked by exploration_editor or
+                    // skill_editor, topic_editor. 
+                    var currentPage = $window.location.pathname.split('/')[1];
```

Comment:
I don't think this is a correct way to handle this, we are going to use this directive in either of two places "exploration editor" or "question editor" page.
I have two ideas:
- Use StateEditorService.isInQuestionMode()
- Populate the state-editor directive with interaction categories (as state editor directive is used in two places so we can populate data as such required.).

I'm not sure about the later idea as we will be injecting this data to multiple layers (From page controller to state-editor directive --> state-interaction directive).

@seanlip any suggestion on which would be a better idea?

Reply:
I think both of these would work. The first one seems simple and the second one is OK as long as it doesn't result in a long chain of argument-passing.

I agree that this logic does not belong in this file.

Reply:
Done.